samcli/commands/build/build_context.py

```diff
@@ -270,6 +270,8 @@ def run(self):
 
             for f in self.get_resources_to_build().functions:
                 EventTracker.track_event("BuildFunctionRuntime", f.runtime)
+                if f.metadata and f.metadata.get("BuildMethod", "") == "esbuild":
+                    EventTracker.track_event("UsedFeature", "ESBuild")
```

Comment:
a question - Did we anticipate that `EventTracker.track_event` would become the point of failure? i.e. Will it cause the command to fail?

Reply:
I don't think we ever did... One solution I can think of might be to put the method in a try-catch? That way instead of throwing an error and breaking the program flow, it could print to console or `LOG`